blueman/main/Tray.py

```diff
@@ -10,16 +10,18 @@
 from gi.repository import GLib, Gio
 
 
-class BluemanTray(object):
+class BluemanTray(Gio.Application):
     def __init__(self):
-        check_single_instance("blueman-tray")
```

Comment:
I guess `GApplication` takes care of this (via the `application_id`), right?

Reply:
`StatusIcon._on_plugins_changed` does not work without this (as the lockfile does not get written). It needs to use the quit action now.

Reply:
The application id is used to identify the app. Then in ``do_activate`` it can do anything you like one of the options is to only allow one instance.

I'll add a proxy like appletservice and make it easy to quit the tray.